mllib/src/main/scala/org/apache/spark/mllib/stat/test/AndersonDarlingTest.scala

```diff
@@ -0,0 +1,278 @@
+      rawCriticalValuess.map { case (sig, cv) => sig -> cv / (1 + 0.2 / math.sqrt(n)) }
+    }
+  }
+
+  /**
+   * Perform a one sample Anderson-Darling test
+   * @param data data to test for a given distribution
+   * @param distName name of theoretical distribution: currently supports normal,
+   *            exponential, gumbel, logistic, weibull as
+   *            ['norm', 'exp', 'gumbel', 'logistic', 'weibull']
+   * @param params variable-length argument providing parameters for given distribution. When none
+   *               are provided, default parameters appropriate to each distribution are chosen. In
+   *               either case, critical values reflect adjustments that assume the parameters were
+   *               estimated from the data
+   * @return
+   */
+  @varargs
+  def testOneSample(data: RDD[Double], distName: String, params: Double*)
+    : AndersonDarlingTestResult = {
+    val n = data.count()
+    val dist = initDist(distName, params)
+    val localData = data.sortBy(x => x).mapPartitions(calcPartAD(_, dist, n)).collect()
+    val s = localData.foldLeft((0.0, 0.0)) { case ((prevStat, prevCt), (rawStat, adj, ct)) =>
+      val adjVal = 2 * prevCt * adj
+      val adjustedStat = rawStat + adjVal
+      val cumCt = prevCt + ct
+      (prevStat + adjustedStat, cumCt)
+    }._1
+    val ADStat = -1 * n - s / n
+    val criticalVals = dist.getCriticalValues(n)
+    new AndersonDarlingTestResult(ADStat, criticalVals, NullHypothesis.OneSample.toString)
+  }
+
+
+  /**
+   * Calculate a partition's contribution to the Anderson-Darling statistic.
+   * In each partition we calculate 2 values, an unadjusted value that is contributed to the AD
+   * statistic directly, a value that must be adjusted by the number of values in the prior
+   * partitions, and a count of the elements in that partition
+   * @param part a partition of the data sample to be analyzed
+   * @param dist a theoretical distribution that extends the AndersonDarlingTheoreticalDist trait,
+   *             used to calculate CDF values and critical values
+   * @param n the total size of the data sample
+   * @return The first element corresponds to the position-independent contribution to the
+   *         statistic, the second is the value that must be scaled by the number of elements in
+   *         prior partitions, and the third is the number of elements in this partition
+   */
+  private def calcPartAD(part: Iterator[Double], dist: AndersonDarlingTheoreticalDist, n: Double)
+    : Iterator[(Double, Double, Double)] = {
+    val initAcc = (0.0, 0.0, 0.0)
+    val pResult = part.zipWithIndex.foldLeft(initAcc) { case ((prevS, prevC, prevCt), (v, i)) =>
+      val y = dist.cdf(v)
+      val a = math.log(y)
+      val b = math.log(1 - y)
+      val unAdjusted = a * (2 * i + 1) + b * (2 * n - 2 * i - 1)
+      val adjConstant = a - b
+      (prevS + unAdjusted, prevC + adjConstant, prevCt + 1)
+    }
```

Comment:
I tried to add an ut for gumbel distribution. yet the number I get from scipy is different from the value calculated here. I'm not sure if the expression here is effective for the asymmetrical distribution. I will investigate more and any comment will be welcome.